examples/src/examples/graphics/integer-textures.mjs

```diff
@@ -0,0 +1,600 @@
+            }
+        });
+
+        // Create a 2D screen for help text
+        const screen = new pc.Entity();
+        screen.addComponent("screen", {
+            referenceResolution: new pc.Vec2(1280, 720),
+            scaleBlend: 0.5,
+            scaleMode: pc.SCALEMODE_BLEND,
+            screenSpace: true
+        });
+        app.root.addChild(screen);
+
+        // Help text
+        const textMarkup = new pc.Entity();
+        textMarkup.setLocalPosition(0, 50, 0);
+        textMarkup.addComponent("element", {
+            alignment: new pc.Vec2(1.0, 0.0),
+            pivot: new pc.Vec2(0.0, 0.0),
+            anchor: new pc.Vec4(0.25, 0.01, 0.96, 0.99),
+            fontAsset: assets.font.id,
+            fontSize: 16,
+            text: 'Left click:    Add\nShift click: Remove\nPress Space:  Reset',
+            width: 500,
+            height: 100,
+            autoWidth: false,
+            autoHeight: false,
+            wrapLines: true,
+            enableMarkup: true,
+            type: pc.ELEMENTTYPE_TEXT
+        });
+        screen.addChild(textMarkup);
+
+        let passNum = 0;
+        app.on("update", function (/** @type {number} */) {
+
+            mouseUniform[0] = mousePos.x;
+            mouseUniform[1] = mousePos.y;
+
+            const brushRadius = data.get('options.brushSize') / Math.max(TEXTURE_WIDTH, TEXTURE_HEIGHT);
+            const brush = data.get('options.brush') ?? 1;
+
+            // Run the sand simulation shader
+            for (let i = 0; i < STEPS_PER_FRAME; i++) {
+                device.scope.resolve('sourceTexture').setValue(sourceTexture);
+                device.scope.resolve('mousePosition').setValue(mouseUniform);
+                device.scope.resolve('mouseButton').setValue(mouseState);
+                device.scope.resolve('brush').setValue(brush);
+                device.scope.resolve('brushRadius').setValue(brushRadius);
+                device.scope.resolve('passNum').setValue(passNum);
+                device.scope.resolve('randomVal').setValue(Math.random());
+                pc.drawQuadWithShader(device, sandRenderTarget, sandShader);
+                device.copyRenderTarget(sandRenderTarget, sourceRenderTarget, true, false);
+                passNum = (passNum + 1) % 16;
+            }
+
+            // Render a visual representation of the simulation
+            device.scope.resolve('sourceTexture').setValue(sandRenderTarget.colorBuffer);
+            device.scope.resolve('mousePosition').setValue(mouseUniform);
+            device.scope.resolve('brushRadius').setValue(brushRadius);
+            pc.drawQuadWithShader(device, outputRenderTarget, outputShader);
+
+        });
+    });
+    return app;
+}
+
+export class IntegerTextureExample {
+    static CATEGORY = 'Graphics';
+    static WEBGPU_ENABLED = true;
+    static WEBGL1_DISABLED = true;
+    static example = example;
+    static controls = controls;
+    static sharedShaderChunks = {
```

Comment:
could this also be part of FILES so that the built in code editor displays this file?

Reply:
ah actually that is not needed, you inlined this into the shaders directly.